src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionIterator.java

```diff
@@ -209,6 +225,26 @@ private void fillNextAssemblyRegionWithReads( final AssemblyRegion region ) {
         }
     }
 
+    private void fillNextAssemblyRegionWithPileupData(final AssemblyRegion region){
```

Comment:
I'm torn... its probably fine to have the assembly engine also hold onto all of the pileups for itself but i'm a little worried this might measurably rise the memory footprint of our variant callers...

Reply:
I fixed this by making this only trigger when we are in pileup mode thus saving us any potential headaches for the standard use case.

Reply:
Hmm.... this is the one pice that is somewhat onerous to fix properly. The issue is this is owned by the AssemblyRegionIterator which has its own (caller-agnostic though its only actually used by M2 and HC) set of arguments so in order to configure this properly i would need to push an argument into there (which doesn't seem like the correct place to put assembly arguments...). The risk here is the memory footprint of all those extra pileup objects not being freed from the heap during the processing of the 2 assembly regions that are in memory at a time. Its possibly not a problem? Opinions?

Reply:
okay I got around this by adding a configuration to the assembly region walker that asks the tool whether to track the pileups.